File: src/main/java/com/chattriggers/ctjs/imports/gui/ImportGui.java
package com.chattriggers.ctjs.imports.gui;

import com.chattriggers.ctjs.imports.Import;
import net.minecraft.client.gui.GuiScreen;

public class ImportGui extends GuiScreen {
    private Import theImport;

    public ImportGui(Import theImport) {
        this.theImport = theImport;
    }

    @Override
    public void drawScreen(int mouseX, int mouseY, float partialTicks) {
        super.drawScreen(mouseX, mouseY, partialTicks);

        theImport.getLines();
    }
}

File: src/main/kotlin/com/chattriggers/ctjs/engine/module/import/ImportGui.kt
package com.chattriggers.ctjs.engine.module.import

import com.chattriggers.ctjs.minecraft.libs.FileLib
import com.chattriggers.ctjs.print
import com.google.gson.*
import net.minecraft.client.gui.GuiScreen
import org.jsoup.nodes.Document
import java.net.UnknownHostException

object ImportGui : GuiScreen() {
    val gson: Gson = GsonBuilder().apply {
        setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
        registerTypeAdapter(Boolean::class.java, BooleanTypeAdapter())
        registerTypeAdapter(Document::class.java, DocumentTypeAdapter())
    }.create()

    var modules: List<ForeignModule> = listOf()

    override fun initGui() {
        updateModulesList()
    }

    private fun updateModulesList() = try {
        modules = gson.fromJson(
                FileLib.getUrlContent("https://www.chattriggers.com/api/modules"),
                Array<ForeignModule>::class.java
        ).toList()
    } catch (e: UnknownHostException) {
        e.print()
    }
}